app/routes/_landing.($lang).docs/route.tsx

```diff
@@ -30,31 +30,63 @@ export const loader: LoaderFunction = async ({ params }) => {
   }
 }
 
-const replaceAll = (str: string, find: string, replace: string) => {
-  return str.replace(new RegExp(find, "g"), replace)
-}
-
 export default function DocsLayout() {
   const { data }: LoaderData = useLoaderData()
   const [linksGroupItems, setLinksGroupItems] = useState<LinksGroupProps[]>([])
 
   useEffect(() => {
     if (data && data.length) {
-      setLinksGroupItems(
-        data.map((doc: documentation) => {
-          return {
-            label:
-              doc.translations && doc.translations[0]?.title
-                ? doc.translations[0].title
-                : "",
-            link: doc.slug ? doc.slug : "",
-            slug: doc.slug ? doc.slug : "",
-          }
-        }),
-      )
+      const organizedData = organizeData(data)
+      setLinksGroupItems(organizedData)
     }
   }, [data])
```

Comment:
`organizeData` is missing from the dependency array. We can memoize the function and pass it here, or a more common approach it can be assigned to a ref and used via the ref which wouldn't be require it to be added to the dependency array.